dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -100,42 +111,63 @@ class PovertyTrapModel(Model):
                             'deletion_prob':0.05,
                             'ratio':0.1,
                             'homophily_parameter':0.69,
-                            'characteristic_distance':35, 
+                            'characteristic_distance':35,
                             'truncation_weight':1.0e-10,}
     }
 
-    def __init__(self,*, model_identifier=None, restart=False, savestate=None):
+    def __init__(self,*, model_identifier, restart=None, savestate=True):
         """
-        restore from a savestate (TODO) or create a PVT model instance.
+        restore from a savestate or create a PVT model instance.
         Checks whether a model indentifier has been specified.
+
+        param: model_identifier: str, required. Identifier for the model. Used to save and load model states.
+        param: restart: int, optional. If specified, the model is run from
+        this step. Default None.
+        param: savestate: bool, optional. If True, the states are saved to files. Default True.
         """
-        if restart:
-            if savestate==None:
-                raise ValueError('When restarting a simulation an intial savestate must be supplied')
-            else:
-                #TODO implement restart
-                pass
-        else:
-            super().__init__(model_identifier = model_identifier)
-            if self._model_identifier == None:
-                raise ValueError('A model identifier must be specified')
-            self.number_agents = None
-            self.gamma_vals = None
-            self.sigma_dist = None
-            self.cost_vals = None
-            self.technology_levels = None
-            self.technology_dist = None
-            self.a_theta_dist = None
-            self.sensitivity_dist = None
-            self.capital_dist = None
-            self.alpha_dist = None
-            self.lambda_dist = None 
-            self.initial_graph_type = None
-            self.model_graph = None
-            self.step_count = None
-            self.step_target = None
-            self.steering_parameters = None
-            self.model_data  = None
+        super().__init__(model_identifier = model_identifier)
+
+        self.restart = restart
+        if self.restart == 0:
+            raise ValueError('restart step should be greater than 0. '
+                             'If you want to start from the beginning, '
+                             'do not specify restart step and run intialize_model()')
+
+
+        if self.restart:
+            path_model_graph = f'./{self._model_identifier}/model_graphs.bin'
+            path_generator_state = f'./{self._model_identifier}/generator_state.bin'
```

Comment:
I like the idea of storing the generator states

Reply:
This is needed to control the randomness after re-starting the model.